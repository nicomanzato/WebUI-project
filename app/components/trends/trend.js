import React from 'react';
import {
  StyleSheet,
  ScrollView,
  View,
  Text,
  TouchableHighlight,
} from 'react-native';

export const Trend = (props) => {

  const tw_volume = props.trend.tweet_volume === null ? null :
   <Text style={styles.volumeText}>{props.trend.tweet_volume} Tweets</Text>;

  return (
    <TouchableHighlight
       key={'t' + props.index}
       onPress={() => { props.onPress(props.trend)}}
       underlayColor='#dddddd'>
       <View style={styles.trend} key={props.index}>
          <View style={styles.trendNumberView}>
            <Text key={'#' + props.index} style={styles.trendNumber}>{props.index + 1}</Text>
          </View>
          <View style={styles.trendTextView}>
            <Text key={props.index + 'Name'} style={styles.trendText}>{props.trend.name}</Text>
            {tw_volume}
          </View>
      </View>
    </TouchableHighlight>
  );

}

export default Trend

const styles = StyleSheet.create({
  trend: {
    borderBottomWidth: 1,
    borderColor: "#ccc",
    flexDirection: 'row',
    padding: 10,

  },
  trendText: {
    fontSize: 17,
    marginBottom: 5
  },
  trendTextView: {
    flex: 0.9,
    paddingLeft: 10,
  },
  trendNumber: {
    fontSize: 18,
    color: 'grey'
  },
  trendNumberView: {
    flex: 0.1,
    paddingLeft: 7,
  },
  volumeText: {
    fontSize: 15,
    color: 'grey',
  }
});
