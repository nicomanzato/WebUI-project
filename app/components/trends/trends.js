'use strict';

import React, { Component } from 'react';
import {
  StyleSheet,
  FlatList,
  View,
  Text,
  ActivityIndicator
} from 'react-native';

import TrendList from './trendList';

class Trends extends Component {

  componentDidMount = () => {
    this.props.onComponentMount();
  }

  componentWillUnmount = () => {
    //this.props.onComponentWillUnmount();
  }

  render = () => {
      return (
       <View>
         <Text style={styles.trendTitle}>Trends for you</Text>
         <View style={styles.separator}/>
         <TrendList
           trends={this.props.trends}
           onItemPress={this.props.onTrendPress}
         />
       </View>
      )
  }
};

export default Trends;

const styles = StyleSheet.create({
  activityIndicatorContainer:{
    backgroundColor: "#fff",
    alignItems: 'center',
    justifyContent: 'center',
    flex: 1,
  },

  timeline: {
    flex:1,
    backgroundColor: '#FFFFFF',
    paddingTop:20
  },
  trendTitle: {
    fontSize: 20,
    fontWeight: 'bold',
    marginLeft: 10,
    marginBottom: 15
  },
  separator: {
    height: 1,
    backgroundColor: '#dddddd'
  },
});
