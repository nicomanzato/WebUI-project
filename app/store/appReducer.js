'use strict';

import { combineReducers } from 'redux'
import PostReducer from './post/postReducer'
import TrendReducer from './trend/trendReducer'
import UserReducer from './user/userReducer'
import nav from './navigation/navReducer'
import ConfigurationReducer from './configuration/configurationReducer'

const AppReducer = combineReducers({
  PostReducer,
  TrendReducer,
  UserReducer,
  nav,
  ConfigurationReducer,
});

export default AppReducer;
