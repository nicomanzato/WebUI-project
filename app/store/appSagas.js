import { call, all, put, takeEvery, takeLatest, take, select } from 'redux-saga/effects'
import { delay } from 'redux-saga'

import postSagas from './post/postSagas'
import trendSagas from './trend/trendSagas'
import userSagas from './user/userSagas'

export default function* appSagas() {
  yield all([
    postSagas(),
    trendSagas(),
    userSagas(),
  ])
}
