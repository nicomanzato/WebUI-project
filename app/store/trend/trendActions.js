export const REQUEST_TRENDS_LOAD = 'REQUEST_TRENDS_LOAD'
export const SUCCESS_TRENDS_LOAD = 'SUCCESS_TRENDS_LOAD'
export const FAILURE_TRENDS_LOAD = 'FAILURE_TRENDS_LOAD'
export const RESET_TRENDS = 'RESET_TRENDS'

export function requestTrendsLoad() {
  return { type: REQUEST_TRENDS_LOAD }
}

export function successTrendsLoad(data) {
  return { type: SUCCESS_TRENDS_LOAD, data: data }
}

export function failureTrendsLoad(er) {
  return {type: FAILURE_TRENDS_LOAD, errorDetail: er}
}

export function resetTrends() {
  return {type: RESET_TRENDS}
}
