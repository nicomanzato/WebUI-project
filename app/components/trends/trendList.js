'use strict';

import React from 'react';
import {
  StyleSheet,
  ScrollView,
  View,
  Text,
  TouchableHighlight,
} from 'react-native';
import {Trend} from './trend'

class TrendList extends React.Component {

  render = () => {

    let trends = [];

    if (this.props.trends.length > 0) {
      trends = this.props.trends.map((trend, index) => {
        return (
         <Trend onPress={this.props.onItemPress} trend={trend} key={index} index={index} />
        );
      });
    }

    return (
      <ScrollView>
        {trends}
      </ScrollView>
    )
  };
}

export default TrendList;

const styles = StyleSheet.create({
  trend: {
    borderBottomWidth: 1,
    borderColor: "#ccc",
    flexDirection: 'row',
    padding: 10,

  },
  trendText: {
    fontSize: 17,
    marginBottom: 5
  },
  trendTextView: {
    flex: 0.9,
    paddingLeft: 10,
  },
  trendNumber: {
    fontSize: 18,
    color: 'grey'
  },
  trendNumberView: {
    flex: 0.1,
    paddingLeft: 7,
  },
  volumeText: {
    fontSize: 15,
    color: 'grey',
  }
});
