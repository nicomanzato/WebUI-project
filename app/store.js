import { applyMiddleware } from 'redux';
import createSagaMiddleware from 'redux-saga'
import appReducer from './store/appReducer'; //Import the reducer
import appSagas from './store/appSagas';
import reactotron from '../ReactotronConfig'

const sagaMonitor = reactotron.createSagaMonitor()


const sagaMiddleware = createSagaMiddleware({ sagaMonitor })

export const store = reactotron.createStore(appReducer, applyMiddleware(sagaMiddleware));

sagaMiddleware.run(appSagas)
