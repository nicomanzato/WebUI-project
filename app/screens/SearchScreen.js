import React from 'react';
import { StyleSheet, Text, View, Button, Animated, Easing, ActivityIndicator } from 'react-native';
import PropTypes from 'prop-types';
import {StackNavigator} from 'react-navigation';
import SearchForm from './../components/search/searchForm'
import TrendList from './../components/trends/trendList';
import PostList from './../components/post/postList'
import {bindActionCreators} from 'redux';
import { connect } from 'react-redux';

import Fade from './../components/animation/fade'

import {
  requestSearchForPost,
  requestSearchMorePost,
  resetPostSearch} from '../store/post/postActions';
import {requestTrendsLoad, resetTrends} from '../store/trend/trendActions';

export class SearchScreen extends React.Component {

  constructor(props) {
    super(props)

    this.state = {
      searchKeyword: '',

      shouldDisplayTrends: true,
      shouldDisplaySearchResults: false,
    };
  }

  componentDidMount = () => {
    this.props.requestTrendsLoad();
  }

  setSearchKeyword = (keyword) => {
    this.setState({
      searchKeyword: keyword,
    })
  }

  handleOnTrendPress = (trend) => {
    if (trend.name !== this.props.searchKeyword) {
      this.props.requestSearchForPost(trend.query);
      this.setSearchKeyword(trend.name);
    }
  }

  handleOnSearchFormReset = () => {
    this.props.resetPostSearch();
    this.props.resetTrends();
    this.props.requestTrendsLoad();
    this.setSearchKeyword('');
  }

  handleOnSubmitSearchForm = (keyword) => {
    if (keyword !== this.props.searchKeyword) {
      this.props.requestSearchForPost(keyword);
    }
  }

  handleOnDoneFadingOutSearchResults = () => {
    if (!this.props.hasSearched){
      this.setShouldDisplaySearchResults(false);
      this.setShouldDisplayTrends(true);
    }
  }

  handleOnDoneFadingOutTrends = () => {
    this.setShouldDisplayTrends(false);
    this.setShouldDisplaySearchResults(true);
  }

  setShouldDisplayTrends = (value) => {
    this.setState({
      shouldDisplayTrends: value,
    });
  }

  setShouldDisplaySearchResults = (value) => {
    this.setState({
      shouldDisplaySearchResults: value,
    });
  }

  render = () => {

    return (
      <View style={styles.container}>
        <SearchForm
          animated
          hasSearched={this.props.hasSearched}
          onSearchInputTextChange={this.setSearchKeyword}
          searchValue={this.state.searchKeyword}
          onReset={this.handleOnSearchFormReset}
          onSubmit={(keyword) => { this.handleOnSubmitSearchForm(keyword) }} />
        {(this.props.loadingTrends || this.props.loadingSearch) &&
          <View style={styles.activityIndicatorContainer}>
            <ActivityIndicator animating={true}/>
          </View>
        }
        {this.state.shouldDisplayTrends && !this.props.loadingTrends &&
          <Fade
            visible={!this.props.hasSearched && !this.props.loadingTrends}
            onDoneFadingOut={this.handleOnDoneFadingOutTrends}>

            <View>
              <Text style={styles.trendTitle}>Trends for you</Text>
              <View style={styles.separator}/>
              <TrendList
                trends={this.props.trends}
                onItemPress={(trend) => { this.handleOnTrendPress(trend) }}
              />
            </View>

          </Fade>
        }
        {this.state.shouldDisplaySearchResults &&
          <Fade
            visible={this.props.hasSearched && !this.props.loadingSearch}
            onDoneFadingOut={this.handleOnDoneFadingOutSearchResults}>
            <PostList
              navigation={this.props.navigation}
              data={this.props.searchResult}
              onEndReached={this.props.requestSearchMorePost}
            />
          </Fade>
        }
      </View>
    );
  };

  static navigationOptions = {
    header: null,
  };

}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#FFFF'
  },
  activityIndicatorContainer:{
    backgroundColor: "#fff",
    alignItems: 'center',
    justifyContent: 'center',
    flex: 1,
  },
  trendTitle: {
    fontSize: 20,
    fontWeight: 'bold',
    marginLeft: 10,
    marginBottom: 15
  },
  separator: {
    height: 1,
    backgroundColor: '#dddddd'
  },
});

const mapStateToProps = (state) => {
    return {
        loadingTrends: state.TrendReducer.loadingTrends,
        trends: state.TrendReducer.trends,

        searchKeyword: state.PostReducer.searchKeyword,
        hasSearched: state.PostReducer.hasSearchedForPost,
        loadingSearch: state.PostReducer.loadingSearchPost,
        searchResult: state.PostReducer.searchPost,
        loadingMoreSearchResults: state.PostReducer.loadingMoreSearchPost,
    }
}

const mapDispatchToProps = {
  requestTrendsLoad: () => requestTrendsLoad(),
  requestSearchForPost: (searchKeyword) => requestSearchForPost(searchKeyword),
  requestSearchMorePost: () => requestSearchMorePost(),
  resetPostSearch: () => resetPostSearch(),
  resetTrends: () => resetTrends(),
}

export default connect(mapStateToProps, mapDispatchToProps)(SearchScreen);
