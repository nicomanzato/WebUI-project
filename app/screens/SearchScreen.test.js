import React from 'react';
import {SearchScreen} from './SearchScreen'
import ReduxWrappedSearchScreen from './SearchScreen'
import { Provider } from 'react-redux';
import PostMock from './../store/post/mock/postMock'
import TrendMock from './../store/trend/mock/trendMock'
import {Trend} from './../components/trends/trend'
import { shallow, mount, ReactWrapper } from 'enzyme';
import {store} from './../store'
import {TouchableHighlight} from 'react-native';

describe('search screen', () => {

  it('should render trends correctly', () => {
    const mockNavigationProp = {state: {params: {data: '1'}}}
    const wrapper = shallow(
      <SearchScreen
        navigation={mockNavigationProp}
        loadingTrends={false}
        trends={[TrendMock, TrendMock, TrendMock, TrendMock]}
        requestTrendsLoad={() => {}}

        searchKeyword={''}
        hasSearched={false}
        loadingSearch={false}
        searchResult={{}}
        loadingMoreSearchResults={false}
      />
     );

     expect(wrapper).toMatchSnapshot();
  });

  it('should render activity indicator correctly', () => {
    const mockNavigationProp = {state: {params: {data: '1'}}}
    const wrapper = shallow(
      <SearchScreen
        navigation={mockNavigationProp}
        loadingTrends={true}
        trends={{}}
        requestTrendsLoad={() => {}}

        searchKeyword={''}
        hasSearched={false}
        loadingSearch={false}
        searchResult={{}}
        loadingMoreSearchResults={false}
      />
     );

     expect(wrapper).toMatchSnapshot();
  });

  it('should render trends correctly when wrapped with redux', () => {
    const mockNavigationProp = {state: {params: {data: '1'}}}
    const wrapper = shallow(
      <ReduxWrappedSearchScreen
        store={store}
        navigation={mockNavigationProp}
        loadingTrends={false}
        trends={[TrendMock, TrendMock, TrendMock, TrendMock]}
        requestTrendsLoad={() => {}}

        searchKeyword={''}
        hasSearched={false}
        loadingSearch={false}
        searchResult={{}}
        loadingMoreSearchResults={false}
      />
     );
    expect(wrapper).toMatchSnapshot();
  });

  it('should handle dispatching its actions', () => {

    const requestTrendsLoad = jest.fn().mockName('mockedFunction');
    const requestSearchForPost = jest.fn().mockName('mockedFunction');
    const requestSearchMorePost = jest.fn().mockName('mockedFunction');
    const resetPostSearch = jest.fn().mockName('mockedFunction');
    const resetTrends = jest.fn().mockName('mockedFunction');

    const mockNavigationProp = {state: {params: {data: '1'}}}
    const wrapper = mount(
      <SearchScreen
        navigation={mockNavigationProp}
        requestTrendsLoad={requestTrendsLoad}
        requestSearchForPost={requestSearchForPost}
        requestSearchMorePost={requestSearchMorePost}
        resetPostSearch={resetPostSearch}
        resetTrends={resetTrends}
        loadingTrends={false}
        trends={[TrendMock, TrendMock, TrendMock, TrendMock]}
        requestTrendsLoad={() => {}}

        searchKeyword={''}
        hasSearched={false}
        loadingSearch={false}
        searchResult={{}}
        loadingMoreSearchResults={false}
      />
     );

     const searchForm = wrapper.find('SearchForm').first();

     searchForm.props().onSubmit();
     expect(requestSearchForPost).toHaveBeenCalled();

     searchForm.props().onReset();
     expect(resetTrends).toHaveBeenCalled();
     expect(resetPostSearch).toHaveBeenCalled();
     expect(requestSearchForPost).toHaveBeenCalled();

     wrapper.update();

     const trend = wrapper.find('Trend').first().find('TouchableHighlight').first();
     trend.props().onPress();
  });

});
