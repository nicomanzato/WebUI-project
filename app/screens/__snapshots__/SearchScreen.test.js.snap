// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search screen should render activity indicator correctly 1`] = `
<Component
  style={
    Object {
      "backgroundColor": "#FFFF",
      "flex": 1,
    }
  }
>
  <SearchForm
    animated={true}
    hasSearched={false}
    onReset={[Function]}
    onSearchInputTextChange={[Function]}
    onSubmit={[Function]}
    searchValue=""
  />
  <Component
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <ActivityIndicator
      animating={true}
      color="#999999"
      hidesWhenStopped={true}
      size="small"
    />
  </Component>
</Component>
`;

exports[`search screen should render trends correctly 1`] = `
<Component
  style={
    Object {
      "backgroundColor": "#FFFF",
      "flex": 1,
    }
  }
>
  <SearchForm
    animated={true}
    hasSearched={false}
    onReset={[Function]}
    onSearchInputTextChange={[Function]}
    onSubmit={[Function]}
    searchValue=""
  />
  <Fade
    onDoneFadingOut={[Function]}
    visible={true}
  >
    <Component>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 20,
            "fontWeight": "bold",
            "marginBottom": 15,
            "marginLeft": 10,
          }
        }
      >
        Trends for you
      </Text>
      <Component
        style={
          Object {
            "backgroundColor": "#dddddd",
            "height": 1,
          }
        }
      />
      <TrendList
        onItemPress={[Function]}
        trends={
          Array [
            Object {
              "name": "Sofía Herrera",
              "promoted_content": null,
              "query": "%22Sof%C3%ADa+Herrera%22",
              "tweet_volume": 11111,
              "url": "http://twitter.com/search?q=%22Sof%C3%ADa+Herrera%22",
            },
            Object {
              "name": "Sofía Herrera",
              "promoted_content": null,
              "query": "%22Sof%C3%ADa+Herrera%22",
              "tweet_volume": 11111,
              "url": "http://twitter.com/search?q=%22Sof%C3%ADa+Herrera%22",
            },
            Object {
              "name": "Sofía Herrera",
              "promoted_content": null,
              "query": "%22Sof%C3%ADa+Herrera%22",
              "tweet_volume": 11111,
              "url": "http://twitter.com/search?q=%22Sof%C3%ADa+Herrera%22",
            },
            Object {
              "name": "Sofía Herrera",
              "promoted_content": null,
              "query": "%22Sof%C3%ADa+Herrera%22",
              "tweet_volume": 11111,
              "url": "http://twitter.com/search?q=%22Sof%C3%ADa+Herrera%22",
            },
          ]
        }
      />
    </Component>
  </Fade>
</Component>
`;

exports[`search screen should render trends correctly when wrapped with redux 1`] = `
<SearchScreen
  hasSearched={false}
  loadingMoreSearchResults={false}
  loadingSearch={false}
  loadingTrends={false}
  navigation={
    Object {
      "state": Object {
        "params": Object {
          "data": "1",
        },
      },
    }
  }
  requestSearchForPost={[Function]}
  requestSearchMorePost={[Function]}
  requestTrendsLoad={[Function]}
  resetPostSearch={[Function]}
  resetTrends={[Function]}
  searchKeyword=""
  searchResult={Array []}
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
  trends={Array []}
/>
`;
